pkg/fanal/analyzer/language/nodejs/yarn/yarn.go

```diff
@@ -78,8 +101,40 @@ func (a yarnAnalyzer) PostAnalyze(_ context.Context, input analyzer.PostAnalysis
 }
 
 func (a yarnAnalyzer) Required(filePath string, _ os.FileInfo) bool {
-	fileName := filepath.Base(filePath)
-	return fileName == types.YarnLock || fileName == types.NpmPkg
+	dirs, fileName := splitPath(filePath)
+
+	if fileName == types.YarnLock &&
+		containsAny(filePath, "node_modules", ".yarn/unplugged") {
+		return false
+	}
```

Comment:
Can you add comments why you use these conditions?

Reply:
done